modules/services/gpg-agent.nix

```diff
@@ -256,13 +256,25 @@ in {
       '') cfg.sshKeys;
     })
 
+    (mkIf (pkgs.stdenv.isDarwin && cfg.enableSshSupport) {
```

Comment:
Since https://github.com/nix-community/home-manager/pull/2915, there's no need for the platform conditions any more.